### PR TITLE
Release 20.6 (old branch)

### PR DESCRIPTION
**Pull request by wsanchez**
This PR is a Klein 20.6 release.

The primary goal for this release is to ship all work that's currently been done with support for Python 2.7 before we drop support for Python 2.7, so that we leave the 2.7 users with the latest available progress.

In addition to incrementing the version, this PR adds a script for automating the release process.

### CODE REVIEW

**Comment by wsanchez** [comment]
@glyph What is the state of #283, and do you want to land that for Python 2?  My inclination would be not to, because it's a big feature branch, and Python 2 users won't benefit from any bug fixes to it.

**Comment by glyph** [comment]
> @glyph What is the state of #283, and do you want to land that for Python 2? My inclination would be not to, because it's a big feature branch, and Python 2 users won't benefit from any bug fixes to it.

It pretty much works, but it has basically zero test coverage, and I don't have a ton of time to work on it right now.  I would not want to hold up any releases because of it.

If someone were to *really* need such a thing on python 2 after the fact, they can grab the version from that branch.  But I think the main point with py2 at this point is compatibility, so I doubt anyone interested in building a major new database-backed system with Klein would want to start there.

**Comment by glyph** [comment]
> done with support for Python 2.4 before we drop support for Python 2.4, so that we leave the 2.4 users

@wsanchez please tell me you mean 2.7

**Comment by wsanchez** [comment]
> > done with support for Python 2.4 before we drop support for Python 2.4, so that we leave the 2.4 users
> 
> @wsanchez please tell me you mean 2.7

Obviously I meant that we first need to port to 2.4 (but not 2.5)… wait, this is a day late.

**Comment by wsanchez** [comment]
I'm going to drop the idea of getting type hints into this release for Python 2.4 users.

**Comment by wsanchez** [comment]
@hawkowl @glyph @moshez perhaps you want to look over the new release script.

File: NEWS.rst [hunk 1]
```diff
@@ -1,6 +1,36 @@
...
+
+15.3.1 - 2015-12-17
+-------------------
+
+15.2.0 - 2015-11-30
+-------------------
+
+15.1.0 - 2015-07-08
+-------------------
+
```
Comment (wsanchez) [inline NEWS.rst:33]: I don't know if leaving these blank like this makes sense, but none of these releases had any news added.

File: NEWS.rst [hunk 2]
```diff
@@ -1,6 +1,36 @@
 NEWS
 ====
 
+20.6.0 - 2020-06-??
```
Comment (wsanchez) [inline NEWS.rst:4]: Set the date before releasing.

File: NEWS.rst [hunk 3]
```diff
@@ -1,6 +1,36 @@
...
+-------------------
+ * This is the last release of Klein expected to support Python 2.
+ * Python 3.8 is now supported by Klein. [`#303 <https://github.com/twisted/klein/pull/303>`_]
+ * Python 3.4 is no longer supported by Klein. [`#284 <https://github.com/twisted/klein/pull/284>`_]
+ * ``klein.app.subroute`` is now also available as ``klein.subroute``. [`#293 <https://github.com/twisted/klein/pull/293>`_]
+ * Support for forms and sessions. [`#276 <https://github.com/twisted/klein/pull/276>`_]
```
Comment (wsanchez) [inline NEWS.rst:10]: @glyph might want to describe this in some more useful way?

**Comment by wsanchez** [comment]
@twm I was thinking about factoring out `release.py` into it's own repo for reuse, FYI.

**Comment by glyph** [comment]
> @twm I was thinking about factoring out `release.py` into it's own repo for reuse, FYI.

yessssssssssss

File: NEWS.rst (hunk 3 above)
Comment (glyph) [reply NEWS.rst:11]: I'm pretty sure that was in 19.6?

"New "forms" and "sessions" subsystems provide official support for POST requests, including CSRF protection, form generation to include CSRF tokens, dependency injection to populate parameters from both the request and session, as well as lightweight JSON API support."

Do you want more than that?

Comment (wsanchez) [reply NEWS.rst:11]: It does appear to be in 19.6.  I'll note it there.

**Comment by wsanchez** [comment]
Renamed the branch, which is now PR #353

**Pull request closed without merging**